### ISSUE TITLE
Is this a working project?

### ISSUE THREAD

**Issue by emmggi** [ClemensElflein/xLocalization#1]
I can only see gerber files, no code for this.

Is this project in development, do you have anything functional right now? How does the indoor positioning work?

Thanks.

**Comment by ClemensElflein** [comment]
Hi there!
I have a test setup running, but I was not able to document my findings yet, since I'm working on other projects in parallel.

I'm getting indoor positions with 10 Hz and approx 10 cm accuracy.

What accuracy / update rate are you looking for? May I ask which project you are working on?

**Comment by emmggi** [comment]
I'm not working on anything yet, I'm collecting data and planning. 
10cm accuracy sounds great.

I'm thinking of something on a bigger scale, not quite indoors. A grass cutter for large surfaces. Can you share your setup? Are you working on the ToF principle with something like NRF or ultrasonic modules?

Thanks.

**Comment by ClemensElflein** [comment]
That is interesting. I started with these modules for the same thing ;-) But unfortunately the range was not enough for our lawn. We would have needed so many modules, that RTK GPS was simply cheaper and more accurate in the end.

If you are working on a robotic lawn mower, I have great news for you: I am building one myself right now.

Last summer, I have built the prototype (and it already works!!) and during winter I started to develop some PCBs so that anyone can easily convert an off-the-shelf mower. Check out these repos for more info:
https://github.com/ClemensElflein/OpenMower
https://github.com/ClemensElflein/open_mower_ros


The PCBs are already done and I'll have them in the next days. Then I'll post updates and building instructions.

**Comment by ClemensElflein** [comment]
As for your initial question: Yes, the modules work with ToF, therefore I expected them to be more accurate. I also noticed that orientation changes have large impact on the computed positions. I'm not really sure why, since it's not based on field strength..

For now, I have installed the official PANS firmware (you can get it here https://www.decawave.com/dwm1001-dw10001-dev-and-mdek1001-documents-source-code-android-application-firmware-image/) on the modules and I'm using the Android app shipped with their documentation to configure the modules and read the position.

This works great, but accuracy and range is a problem for me. Therefore I currently don't have any NodeMCU code yet.

**Comment by emmggi** [comment]
So you ended up using an RTK GPS module for positioning? Interesting. How does it fare with cloudy weather and what module is it? I can tell it's a ublox but if it's something expensive then that's out of my reach.

Robotic lawn mowers aren't the only application I thought about that needs a positioning system.

I may attempt to create my own positioning system for smaller scales with the `nrf24l01`s, got tips?

**Comment by ClemensElflein** [comment]
Correct. We are using the Ardusimple RTK module. You can get it here: https://www.ardusimple.com/rtk-starter-kits/. The most basic kit is 211 Euros. Maybe there is also another breakout board available, but the module itself seems to be quite expensive. Therefore I think that the price is fair.

We are using one board for positioning in the robot and another one as base station. You can however stream the RTK data from the internet. Then one kit will be enough.

I did not notice any issues with weather. It's very stable.

With the NR24 you'd use the field strength to do positioning I guess. I haven't tried it and I don't like this approach, since obstacles in the LOS will greatly influence the measured positions.

**Comment by emmggi** [comment]
Thank you. I only found ublox RTK chips so far but they're out of reach for me. The only affordable one would be the NEO-6M which isn't nearly as precise as it needs to be.

In theory, a neural network could be trained to cancel out any large oscillations of the position, as well as using other transceivers that don't have obstacles inbetween.
